aws_lambda_builders/workflows/go_dep/workflow.py

```diff
@@ -0,0 +1,63 @@
+"""
+Go Dep Workflow
+"""
+
+import logging
+import os
+
+from aws_lambda_builders.actions import CopySourceAction
+from aws_lambda_builders.workflow import BaseWorkflow, Capability
+
+from .actions import DepEnsureAction, GoBuildAction
+from .utils import OSUtils
+from .subproc_exec import SubprocessExec
+
+LOG = logging.getLogger(__name__)
+
+class GoDepWorkflow(BaseWorkflow):
+    """
+    A Lambda builder workflow that knows how to build
+    Go projects using `dep`
+    """
+
+    NAME = "GoDepBuilder"
+
+    CAPABILITY = Capability(language="go",
+                            dependency_manager="dep",
+                            application_framework=None)
+
+    EXCLUDED_FILES = (".aws-sam")
+
+    def __init__(self,
+                source_dir,
+                artifacts_dir,
+                scratch_dir,
+                manifest_path,
+                runtime=None,
+                osutils=None,
+                **kwargs):
+
+        super(GoDepWorkflow, self).__init__(source_dir,
+                                            artifacts_dir,
+                                            scratch_dir,
+                                            manifest_path,
+                                            runtime=runtime,
+                                            **kwargs)
+
+        options = kwargs["options"] if "options" in kwargs else {}
+        handler = options.get("handler", None)
+
+        if osutils is None:
+            osutils = OSUtils()
+
+        # project base name, where the Gopkg.toml and vendor dir are.
+        base_dir = osutils.abspath(osutils.dirname(manifest_path))
+        output_path = osutils.joinpath(osutils.abspath(artifacts_dir), handler)
```

Comment:
Does this mean the binary is going to be under `.aws-sam/build/<logical_id>/<handler name>/`?

Reply:
It'll be under whatever we specify as the artifact path in SAM CLI, which is likely to be the above (which is fine since the `CodeUri` property will define the path)

Reply:
The convention is to place the code under the logical_id (not a subfolder). 

Can we match that expectation here?

Reply:
Surely, it's for whatever calls the aws-lambda-builder to define this, not this projects.

There's no guarantee that SAM CLI will always be the sole thing using this project.

Reply:
From what I can see in other builds, they write directly to the artefacts dir, the same way I do.

The go builder does nothing differently, the `output_path` is just `artifacts_dir/handler`

Reply:
Adding a subfolder is breaking the expectation, this has nothing to do with SAM CLI. We should be placing the artifacts in that folder directly not creating a subfolder. As a caller, I would need to understand this. This will actually break SAM CLI because the CodeUri is updated to be what the `artifacts_dir` is and this builder is creating a subfolder to place the artifacts in.

Reply:
`output_path` is the path to the binary not to a folder. I got those mixed up here.